tests/unit/s2n_client_hello_retry_test.c

```diff
@@ -1388,6 +1398,79 @@ int main(int argc, char **argv)
                                    S2N_ERR_BAD_MESSAGE);
      }
 
+     /**
+     *= https://tools.ietf.org/rfc/rfc8446#4.1.2
+     *= type=test
+     *# -  Removing the "early_data" extension (Section 4.2.10) if one was
+     *#    present.  Early data is not permitted after a HelloRetryRequest.
+     **/
```

Comment:
Is this the right place for this check? What about [this test](https://github.com/aws/s2n-tls/blob/a7b97a0dcff52c3efec303a0204c75200963777f/tests/unit/s2n_client_early_data_indication_test.c#L95-L115)?

if this is the right place, then you need more context in the comment. What is this from a list of? Where is the MUST?

Reply:
I didn't see the test you linked originally, and it definitely seems better! I removed this new test and added the compliance comment to the one you linked: 6a4e521